daprdocs/content/en/operations/components/pluggable-components/pluggable-components-registration.md

```diff
@@ -0,0 +1,228 @@
+---
+type: docs
+title: "How-To: Register a pluggable component"
+linkTitle: "How To: Register a pluggable component"
+weight: 4500
+description: "Learn how to register a pluggable component"
+---
+
+[uds]: https://en.wikipedia.org/wiki/Unix_domain_socket
+
+## Component Registration Process
+
+Pluggable, [gRPC-based](https://grpc.io/) components are typically run as containers or processes that need to communicate with the Dapr runtime via [Unix Domain Sockets][uds]. They are automatically discovered and registered in the runtime with the following steps:
+
+1. The component listens to an [Unix Domain Socket][uds] placed on the shared volume.
+2. The Dapr runtime lists all [Unix Domain Socket][uds] in the shared volume.
+3. The Dapr runtime connects with each socket and uses gRPC reflection to discover all proto services from a given building block API that the component implements.
+
+A single component can implement multiple [building blocks]({{< ref building-blocks-concept.md >}}) at once.
+
+<img src="/images/grpc-components.png" width=50%>
+
+While Dapr's built-in components come [are included with the runtime](https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md), pluggable components require a few setup steps before they can be used with Dapr.
+
+1. Pluggable components need to be started and ready to take requests _before_ Dapr itself is started.
+2. The [Unix Domain Socket][uds] file used for the pluggable component communication need to be made accessible to both Dapr and pluggable component.
+
+Dapr does not launch any pluggable components processes or containers. This is something that you need to do, and it will be different depending on how Dapr and your components are run:
+
+- In self-hosted mode as processes or containers.
+- In Kubernetes, as containers.
+
+This also changes the approach to share [Unix Domain Socket][uds] files between Dapr and pluggable components.
+
+{{% alert title="Note" color="primary" %}}
+As a prerequisite the operating system must supports Unix Domain Sockets, any UNIX or UNIX-like system (Mac, Linux, or for local development [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) for Windows users) should be sufficient.
+{{% /alert %}}
+
+Select your environment to begin making your component discoverable.
+
+{{< tabs "Standalone" "Kubernetes" >}}
+
+{{% codetab %}}
+[uds]: https://en.wikipedia.org/wiki/Unix_domain_socket
+
+## Run the component
+
+Both your component and the Unix Socket must be running before Dapr starts.
+
+By default, Dapr looks for [Unix Domain Socket][uds] files in the folder in `/tmp/dapr-components-sockets`.
+
+The name of the file without any extension is the name of the component. For example, for `memstore.sock`, the component name is `memstore`.
```

Comment:
I think we should rewrite this to make the file name creation depend on the component's name. The way it is currently written is the other way around.

```suggestion
Filenames in this folder are significant for the component discovery process. They must be formed by appending the component's name with a file extension of your choice, more commonly `.sock`. For example, the filename `memstore.sock` is a valid UDS file name for a component named `memstore`.
```

Reply:
@tmacam replace “discovery process ” by “register”

Reply:
Updated.